internal/isoeditor/rhcos.go

```diff
@@ -382,6 +404,66 @@ func IgnitionImageArchive(ignitionConfig string) ([]byte, error) {
 	return compressedBuffer.Bytes(), nil
 }
 
+func readHeader(isoPath string) (*OffsetInfo, *OffsetInfo, error) {
```

Comment:
This whole function looks like we're just doing the same thing for each reserved area.
It might be worth creating a single function that takes the 24 raw bytes, parses out the struct, and validates it against a given key string and calling that twice.

That would also be easier to use in the full-iso case. And it would be cleaner to extend if we add additional reserved areas (for kernel args, for example).

Reply:
right, would be useful for the full-iso scenario for reusability. changed to: 'ParseOffsetInfo(headerBytes []byte, key string)'